src/liblsquic/lsquic_full_conn_ietf.c

```diff
@@ -1708,6 +1708,7 @@ lsquic_ietf_full_conn_server_new (struct lsquic_engine_public *enpub,
     return &conn->ifc_conn;
 
   err3:
+    conn->ifc_enpub->enp_stream_if->on_conn_closed(&conn->ifc_conn);
```

Comment:
I think on_conn_closed only be called by ietf_full_conn_ci_destroy would be better, we can change like following:
Call on_new_conn after handshake_ok, it's ok? @linsichen206 @litespeedtech 
<img width="721" alt="image" src="https://user-images.githubusercontent.com/12257285/206669018-a1248c09-c817-4699-b87e-744906271565.png">